projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -232,6 +232,43 @@ function get_subscriber_count( $include_social_followers ) {
 	return $subscriber_count;
 }
 
+/**
+ * Returns the newsletter categories from the .com API or the Jetpack XML-RPC API.
+ *
+ * @return array containing [ 'enabled' => true|false, 'newsletter_categories' => array ]
+ */
+function fetch_newsletter_categories() {
+	$categories = array();
+
+	if ( is_wpcom() ) {
+		$categories = \wpcom_fetch_newsletter_categories();
+	} else {
+		$xml = new \Jetpack_IXR_Client();
+		$xml->query( 'jetpack.fetchNewsletterCategories' );
+
+		if ( ! $xml->isError() ) {
+			$categories = $xml->getResponse();
+		}
+	}
+
+	return $categories;
+}
+
+/**
+ * Returns the newsletter categories if the feature is enabled, otherwise returns an empty array.
+ *
+ * @return array
+ */
+function get_newsletter_categories() {
+	$response = fetch_newsletter_categories();
+
+	if ( isset( $response['enabled'] ) && $response['enabled'] ) {
+		return $response['newsletter_categories'];
```

Comment:
Should we do additional checks and casting to ensure that `$response['enabled']` is a boolean, that `$response['newsletter_categories']` is set and is an array?